src/BrandwatchReactAuth.js

```diff
@@ -0,0 +1,73 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import jwtDecode from 'jwt-decode';
+import TokenStore from 'viziaauth';
+
+export default class BrandwatchReactAuth extends Component {
+  getChildContext() {
+    return {
+      brandwatchAuthLogout: this.handleLogout,
+      brandwatchAuthGetProfile: this.handleGetProfile,
+      brandwatchAuthGetToken: this.handleGetToken,
+    };
+  }
+
+  constructor(props) {
+    super(props);
+    this.store = new TokenStore(this.props.domain),
+    this.handleGetProfile = this.handleGetProfile.bind(this);
+    this.handleGetToken = this.handleGetToken.bind(this);
+    this.handleLogout = this.handleLogout.bind(this);
+
+    this.state = {
+      loggedIn: undefined,
+    };
+
+    props.onCreateStore(this.store);
+  }
+
+  componentWillMount() {
+    this.handleGetToken().then((token) => {
+      if (!token) {
+        return window.location.href = this.store.loginUrl;
+      }
+
+      this.setState(() => ({ loggedIn: true }));
```

Comment:
I may be be wrong but because the redirect inside `componentWillMount` is asynchronous , if there's no conditional render then it will continue to render the app before it has finished the token request. Or am I missing something?